File: backend/models/student.go
package models

import (
	"time"
)

type Student struct {
	ID                    uint64    `json:"id"`
	FirstName             string    `json:"first_name"`
	LastName              string    `json:"last_name"`
	DOB                   time.Time `json:"dob"`
	Gender                string    `json:"gender"`
	AddressLine1          string    `json:"address_line1"`
	AddressLine2          string    `json:"address_line2"`
	City                  string    `json:"city"`
	ContactNumber         string    `json:"contact_number"`
	ContactNumberGuardian string    `json:"contact_number_guardian"`
	SupervisorID          *uint     `json:"supervisor_id"`
	Remarks               string    `json:"remarks"`
	HomeLong              float64   `json:"home_long"`
	HomeLat               float64   `json:"home_lat"`
	EmployerID            *uint     `json:"employer_id"`
	CheckInTime           string    `json:"check_in_time"`
	CheckOutTime          string    `json:"check_out_time"`
}

File: web-app/src/types/student.ts
export interface Student {
  id: number;
  first_name: string;
  last_name: string;
  dob: string; // ISO string
  gender: string;
  address_line1: string;
  address_line2: string;
  city: string;
  contact_number: string;
  contact_number_guardian: string;
  supervisor_id?: number | null;
  remarks: string;
  home_long: number;
  home_lat: number;
  employer_id?: number | null;
  check_in_time: string;
  check_out_time: string;
}
// export interface Student {
//     id: number;
//     first_name: string;
//     last_name: string;
//     dob: Date;
//     gender: string;
//     address_line1: string;
//     address_line2: string;
//     city: string;
//     contact_number: string;
//     contact_number_guardian: string;
//     supervisor_id: number;
//     employer_id: number;
//     remarks: string;
//     photo: Uint8Array;
//     home_long: string;
//     home_lat: string;
// }
